### DISCUSSION TITLE
How to build packages with AUR dependencies in chroot?

### DISCUSSION THREAD

**Discussion by Mushoz** [Morganamilo/paru#919] [Q&A]
As recommended by the AUR ROCM stack, I am trying to build the packages in a clean chroot via:

`paru --chroot -S rocm-hip-sdk rocm-opencl-sdk`

(Mind you, I have followed paru's instruction to enable the local repo)

However, I need to rerun this command multiple times, since packages are not compiled and installed in the correct order as to satisfy the AUR dependencies that these packages have. Every time I invoke the command, it will eventually fail on some packages that do not have their dependencies met, but some packages will successfully build and will therefor be synced to my local repo. This means that the subsequent run of the same command will have more packages successfully build, since more dependencies are now available.

But isn't there a way for paru to build these in the correct order to ensure I only need to run this command once? Right now I have to come back multiple times to my computer to rerun the command to build more packages. I feel like I am missing something, but I am not entirely sure what. Any pointers would be much appreciated!

**Comment by Morganamilo** [comment]
My guess is either this is a bug with paru and fixed in paru-git. Or the toolchain is a bootstrap package?

**Reply by Mushoz** [reply]
What exactly do you mean with "Or the toolchain is a bootstrap package?"

I could definitely compile and try again with paru-git!

**Reply by Morganamilo** [reply]
Like gcc or something where the package order is complex as you need to:

- install bootstrap package
- build some stuff that depends on it
- use that to build the real package 
- remove bootstrap

No idea if it's the case here. I just know tool chains are often like this.

**Reply by Mushoz** [reply]
I am fairly sure that is not the case here. Most packages are build via rocm-llvm, which only has 3 makedeps: ninja, cmake and python. Furthermore, some of them require rocm-cmake which is also trivially built.

The installation of my above command is actually giving a build error based on the regular deps, and NOT on the makedeps. But since some packages do build fine and are synced to the local repo, invoking the command again allows it to continue further next time the command is ran.

I will try again with paru-git as soon as I have access to my desktop again and report back.

**Reply by Mushoz** [reply]
Unfortunately, after repeating the same command over and over again, eventually I got stuck where running it again would not get me any further into the process. So I decided to try paru-git instead. However, that one is also running into similar issues while installing. It's currently failing with the following error:

```
warning: cannot resolve "miopen-hip", a dependency of "rocm-hip-sdk"
:: The following package cannot be upgraded due to unresolvable dependencies:
      rocm-hip-sdk

:: Do you want to skip the above package for this upgrade? [y/N] 
error: failed to prepare transaction (could not satisfy dependencies)
:: unable to satisfy dependency 'miopen-hip' required by rocm-hip-sdk
```

miopen-hip is available on the AUR just fine, but for some reason paru does not build it first to meet the dependency of rocm-hip-sdk. Any ideas?

**Reply by Mushoz** [reply]
@Morganamilo should I open a new issue rather than a discussion? Since it seems like this is turning out to be a bug, instead of a question, right? Please let me know, and I will happily open a new issue to track this problem :)

**Reply by Morganamilo** [reply]
Yeah. Can't say more without full output.

**Reply by Mushoz** [reply]
Sure! I have just created it here, this time with the full output: https://github.com/Morganamilo/paru/issues/920
